src/client.ts

```diff
@@ -0,0 +1,239 @@
+    })
+    .then(() => {
+      console.info('Registered Application Commands successfully.');
+      // TODO: Log
+    })
+    .catch((err) => {
+      // TODO: Log err
+      console.error(`Could not register Application Commands: ${err}`);
+      process.exit(1);
+    });
+}
+
+/**
+ * Checks whether a message is valid or not.
+ *
+ * Checks whether the message is:
+ * - in an enabled channel,
+ * - not posted by a bot user,
+ * - that the message is less than 500 characters.
+ */
+function message_is_valid(message: Message): boolean {
+  return (
+    Config.ENABLED_CHANNEL_IDS.includes(message.channel.id) &&
+    !message.author.bot &&
+    message.content.length <= 500
+  );
+}
+
+/**
+ * Registers Discord callbacks for posted messages and message reactions.
+ *
+ * @param {Client} client - The Discord client.
+ * @param {GameSummaryMessage} game_summary_message - Game summary message to register.
+ */
+function register_callbacks(
+  client: Client,
+  game_summary_message: GameSummaryMessage,
+  commands?: SlashCommand[]
+) {
+  const games = game_summary_message.get_games();
+
+  client.on(Events.MessageCreate, async (message) => {
+    if (message_is_valid(message)) {
+      await Promise.all(games.map((game) => game.handle_message(message)))
+        .then((games) => {
+          if (games.filter((entry) => entry !== undefined).length > 0)
+            console.info(
+              `Found valid game message for games ${games
+                .filter((entry) => entry !== undefined)
+                .map((entry) => `'${entry.game}'`)
+                .join(', ')}.`
+            );
+        })
+        .catch((err) => console.warn(`Message callback failed: ${err}.`));
+    }
+  });
+  console.info(
+    `Registered callbacks for: ${game_summary_message
+      .get_games()
+      .map((g) => g.name)
+      .join(', ')}.`
+  );
+
+  client.on(Events.MessageReactionAdd, async (message_reaction, user) => {
+    const channel_id = message_reaction.message.channel.id;
+    const last_message = message_reaction.message.channel.lastMessage;
+    const bot_id = message_reaction.client.user.id;
+
+    if (
+      Config.ENABLED_CHANNEL_IDS.includes(channel_id) &&
+      !user.bot &&
+      !(last_message?.author.id === bot_id && last_message?.embeds.length > 0)
+    ) {
+      console.info('Valid reaction found, sending summary message.');
+      await game_summary_message.send(message_reaction).catch((err) => {
+        console.error(
+          `Something went wrong while sending game summary message: ${err}`
+        );
+      });
+    }
+  });
+  console.info(`Registered message reaction callback.`);
+
+  if (commands !== undefined) {
+    client.on(Events.InteractionCreate, async (interaction) => {
+      if (!interaction.isChatInputCommand()) return;
+
+      await Promise.all(
+        commands.map((command) =>
+          command.definition.name === interaction.commandName
+            ? command.handler(interaction)
+            : undefined
```

Comment:
I just picked `undefined` because optional parameters seem to do so too, e.g. `a?: string` is equivalent to `a: string | undefined`. Mostly semantic I think?